src/common/conditional_compilation/docs/develop_cc_for_new_component.md

```diff
@@ -0,0 +1,209 @@
+They are very similar, and functionality is same with MATCHER_SCOPE plus throw error.
+
+### 3. Non Conditional Compilation
+It means that conditional compilation is disabled.
+
+#### OV_CC_DOMAINS
+It is empty in this mode
+
+#### OV_SCOPE
+It is empty in this mode
+
+#### OV_SCOPE and OV_CASE
+They will do match operation without itt scope task.
+
+#### MATCHER_SCOPE
+It is empty.
+
+#### RUN_ON_FUNCTION_SCOPE and RUN_ON_MODEL_SCOPE
+They are empty.
+
+
+## Enable conditional complication for a new component or domain
+### step 1: requirement analyze
+Analyze which code region need conditional complication and figure out requirement for conditional complication.
+
+### step 2: create itt.hpp if possible
+Create itt.hpp file in the target component directory, in which you need define some particular macro needed by conditional compilation if existed macro cannot meet requirement. An example [itt.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/itt.hpp).
+
+- Add macro into itt.hpp to define new CC domain:
+```ruby
+    #include <openvino/cc/selective_build.h>
+    #include <openvino/itt.hpp>
+    #include <openvino/cc/ngraph/itt.hpp>
+
+    namespace ov{
+    namespace <your_component> {
+    namespace itt {
+    namespace domains {
+        OV_ITT_DOMAIN(your_domain_name);
+    }
+    }
+    }
+    }
+
+    OV_CC_DOMAINS(your_domain_name);
+```
+    
+- Define new macro according to the actual requirement by target component
+  The same macro should be defined in the 3 modes separately, below is an example:
+
+```ruby
+    #if defined(SELECTIVE_BUILD_ANALYZER)
+
+    #define MY_SCOPE(region) OV_SCOPE(my_op, region)
+
+    #elif defined(SELECTIVE_BUILD)
+
+    #define MY_SCOPE(region) MATCHER_SCOPE_(my_op, region)
+
+    #else
+
+    #define MY_SCOPE(region)
+
+    #endif
+```
+
+### step 3: apply CC macro
+Apply the CC macro into your code region that need conditional compilation.
+An example:
+```ruby
+    shared_ptr<ov::Node> ov::op::internal::AUGRUCell::clone_with_new_inputs(const OutputVector& new_args) const {
+        My_SCOPE(internal_AUGRUCell_clone_with_new_inputs);
+        check_new_args_count(this, new_args);
+        return make_shared<AUGRUCell>(new_args.at(0),
+                                  new_args.at(1),
+                                  new_args.at(2),
+                                  new_args.at(3),
+                                  new_args.at(4),
+                                  new_args.at(5),
+                                  get_hidden_size());
+    }
+```
+   Then code region in ov::op::internal::AUGRUCell::clone_with_new_inputs() is implement conditional compilation. If in the SELECTIVE_BUILD_ANALYZER mode it was not called, then in SELECTIVE_BUILD Mode, the code region in this function will be stripped out. 
+
+
+
+## How to verify the new conditional compilation feature
+
+#### Build OpenVINO in SELECTIVE_BUILD_ANALYZER mode
+```ruby
+    cmake \
```

Comment:
You need to create a build folder

Reply:
Sure, I will put completely command here.